### PR TITLE
feat(color-contrast): add support for CSS mix-blend-mode

### PR DESCRIPTION
**Pull request by Zidious**
`mix-blend-mode` functions:

- [x] unit tests
- [x] integration tests

This PR focuses on blending BG colours. An example of mix-blend-mode in action, consider the following: 

```html
<div id="A" style="background-color: rgba(255, 255, 255, 1.0);">
  <div id="B" style="background-color: rgba(0, 128, 0, 0.25);">
    <div id="C" style="background-color: rgba(255, 0, 0, 0.5); mix-blend-mode: difference;">
      <div id="D" style="background-color: rgba(0, 0, 255, 0.25);">Test Element</div>
    </div>
  </div>
</div>
```

When mix-blend-mode is applied onto a div it is considered an [isolated group](https://drafts.fxtf.org/compositing-1/#isolatedgroups) and therefore its own [stacking context](https://drafts.fxtf.org/compositing-1/#csscompositingrules_CSS). 

The stack of the above snippet would look like the following: 

```
Root Group
  - #A
  - #B
  - isolated group
    - #C 
    - #D
```

When blending we do bottom up: 

```
rootGroup = normal(B,normal(A, rootBackDrop))
isolatedGroup = normal(D, C)
difference(rootGroup, isolatedGroup) // outputs the final colour
```

For reference: [compositing BG colours w3c assistance](https://github.com/w3c/fxtf-drafts/issues/440)

Resources: [Compositing and Blending](https://drafts.fxtf.org/compositing-1/#blending) - Note: further research I would like to investigate how to go about blending nested mix-blend-modes when in the same div block.

Closes issue: https://github.com/dequelabs/axe-core/issues/3187

### CODE REVIEW

File: test/commons/color/flatten-colors.js [hunk 1]
```diff
@@ -38,9 +38,251 @@ describe('color.flattenColors', function() {
     assert.equal(flat5.alpha, 1);
 
     var flat6 = axe.commons.color.flattenColors(quarterLightGreen, halfRed);
-    assert.equal(flat6.red, 96);
```
Comment (dylanb) [inline test/commons/color/flatten-colors.js:41]: why did this change?

File: test/integration/full/contrast/blending.js [hunk 2]
```diff
@@ -1,18 +1,173 @@
...
   var resultElms = [];
   var expected = [
+    // normal
     'rgb(223, 112, 96)',
     'rgb(255, 128, 128)',
     'rgb(191, 223, 191)',
     'rgb(125, 38, 54)',
     'rgb(179, 38, 0)',
     'rgb(179, 0, 77)',
-    'rgb(143, 192, 80)',
```
Comment (dylanb) [inline test/integration/full/contrast/blending.js:11]: why this change? Seems like a rounding difference of some sort.

**Review by dylanb — COMMENTED** [review]
two questions, see comments

File: test/commons/color/flatten-colors.js (hunk 1 above)
Comment (Zidious) [reply test/commons/color/flatten-colors.js:41]: The way were were actually blending the colours was not quite accurate. When you start adding alpha values to colours the way they are calculated is different. Opacity adds another step whereby the pre-multiplied computed new RGB values need to be divided by the new computed alpha, we were missing this step: 

> To extract the color component of a pre-multiplied value, the formula is reversed: Co = co / αo
source: https://www.w3.org/TR/compositing-1/#simplealphacompositing
helpful source: https://github.com/w3c/fxtf-drafts/issues/440#issuecomment-956418953

File: test/integration/full/contrast/blending.js (hunk 2 above)
Comment (Zidious) [reply test/integration/full/contrast/blending.js:11]: I believe this may of impacted by https://github.com/dequelabs/axe-core/pull/3226#discussion_r752386049 and the threshold of rounding was impacted

**Review by dylanb — APPROVED** [review]
